extensions/scarb-cairo-run/src/main.rs

```diff
@@ -61,13 +62,15 @@ fn main() -> Result<()> {
 
     ui.print(Status::new("Running", &package.name));
 
-    let sierra_program = cairo_lang_sierra::ProgramParser::new()
-        .parse(
-            &fs::read_to_string(path.clone())
-                .with_context(|| format!("failed to read Sierra file: {path}"))?,
-        )
-        .map_err(|e| anyhow!("{e}"))
-        .with_context(|| format!("failed to parse sierra program: {path}"))?;
+    let sierra_program = serde_json::from_str::<VersionedProgram>(
+        &fs::read_to_string(path.clone())
+            .with_context(|| format!("failed to read Sierra file: {path}"))?,
+    )
+    .with_context(|| format!("failed to deserialize Sierra program: {path}"))?
+    .into_v1()
+    .with_context(|| {
+        format!("failed to parse file into program - incorrect Sierra version: {path}")
+    })?;
```

Comment:
1. Using `-` to denote causability in errors seems inconsistent, given we have nice contexts mechanism in Anyhow
2. I think we can rely on `into_v1` function giving us the cause message, so maybe let's just do this:

```suggestion
    .into_v1()
    .with_context(|| format!("failed to load Sierra program: {path}"))?;
```